File: examples/unsafe-sinks.js
// Intentional demo of unsafe sanitization
// DO NOT use these patterns in production

import express from "express";
import { exec } from "child_process";

const app = express();

app.get("/render", (req, res) => {
  const html = req.body.html;
  // High: XSS via innerHTML
  document.querySelector("#out").innerHTML = html;
  res.send("done");
});

app.get("/eval", (req, res) => {
  const code = req.body.code;
  // Critical: arbitrary code execution
  eval(code);
  res.send("done");
});

app.get("/exec", (req, res) => {
  const path = req.query.path;
  // Critical: command injection
  exec(`ls ${path}`, (err, stdout) => {
    res.send(stdout);
  });
});

app.get("/search", (req, res) => {
  const q = req.params.q;
  // Critical: SQL injection (pseudo-code)
  db.query(`SELECT * FROM items WHERE name = '${q}'`, (err, rows) => {
    res.send(rows);
  });
});

File: examples/unsafe-sinks.py
# Intentional demo of unsafe sanitization
# DO NOT use these patterns in production

from flask import Flask, request
import subprocess
import os
import sqlite3

app = Flask(__name__)
conn = sqlite3.connect(":memory:")

@app.route("/user")
def get_user():
    user_id = request.args.get("id")
    cursor = conn.cursor()
    # Critical: SQL injection via f-string
    cursor.execute(f"SELECT * FROM users WHERE id = {user_id}")
    return cursor.fetchone()

@app.route("/calc")
def calculate():
    expr = request.args.get("expr")
    # Critical: arbitrary code execution
    return str(eval(expr))

@app.route("/list")
def list_files():
    path = request.args.get("path", ".")
    # Critical: command injection
    subprocess.run(f"ls {path}", shell=True)
    return "done"

@app.route("/hello")
def hello():
    name = request.args.get("name")
    from flask import render_template_string
    # High: SSTI
    return render_template_string(f"Hello {name}")
